### PR TITLE
Creating Api endpoints for mobile and console apps

### PR DESCRIPTION
**Pull request by why-a-en**
### Branch changes

- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post. 
- Implemented API endpoints to receive JSON and respond to JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Api::V1::CommentsController < ApplicationController
+  def index
+    @post = Post.find_by_id(params[:post_id])
+    @comments = @post.comments
+    render json: @comments
+  end
```
Comment (aamirkhan2478) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] If I hit this API `/api/v1/users/2/posts` then give me the following error :point_down: 

Screenshot from 2023-08-05 02-01-44

- [ ] Kindly fix this error and please test your API endpoints on Postman :slightly_smiling_face:

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @post = Post.find_by_id(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+    @comment.author = current_user
+
+    if @comment.save
+      render json: @comment
+    else
+      render json: { errors: @comment.errors.full_messages }
+    end
+  end
```
Comment (aamirkhan2478) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] If I hit this API `/api/v1/users/2/posts/1/comments` then give me the following error :point_down: 

Screenshot from 2023-08-05 02-02-35

- [ ] So kindly fix this error as well :slightly_smiling_face:

**Review by aamirkhan2478 — CHANGES_REQUESTED** [review]
Hi @anbehindY  & @Lul-Abdifan , 



Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort
✔️ Your design is good.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bienvenuushindi — APPROVED** [review]
Hi @anbehindY and @Lul-Abdifan , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## HIGHLIGHTS
- No linters errors :heavy_check_mark: 
- Project document professionally :+1: 
- Good PR title and Summary :100: 
- Descriptive Readme :heavy_check_mark: 

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**